src/FffGcodeWriter.cpp

```diff
@@ -2379,159 +2380,187 @@ bool FffGcodeWriter::addSupportToGCode(const SliceDataStorage& storage, LayerPla
 
-    //Print the thicker infill lines first. (double or more layer thickness, infill combined with previous layers)
-    for(const PathOrderOptimizer<const SupportInfillPart*>::Path& path : island_order_optimizer.paths)
+    for(const auto& path : wall_toolpaths)
     {
-        const SupportInfillPart& part = *path.vertices;
+        if(path.empty())
+            continue;
 
-        // always process the wall overlap if walls are generated
-        const int current_support_infill_overlap = (part.inset_count_to_generate > 0) ? default_support_infill_overlap : 0;
+        added_something = true;
+        // Assume for now the wall_toolpaths are sorted againts idx
+        setExtruder_addPrime(storage, gcode_layer, config.extruder_nr);
+        gcode_layer.setIsInside(false); //Going to print walls, which are always outside.
 
-        // add outline (boundary) if any wall is generated
-        if (!part.insets.empty())
+        for (const auto& wall : path)
         {
-            Polygons all_insets;
-            for (const Polygons& inset : part.insets)
-            {
-                all_insets.add(inset);
-            }
-
-            if (all_insets.size() > 0)
-            {
-                setExtruder_addPrime(storage, gcode_layer, extruder_nr); // only switch extruder if we're sure we're going to switch
-                gcode_layer.setIsInside(false); // going to print stuff outside print object, i.e. support
-                gcode_layer.addPolygonsByOptimizer(all_insets, gcode_layer.configs_storage.support_infill_config[0]);
-            }
+            gcode_layer.addWall(wall, gcode_layer.configs_storage.support_infill_config[0]);
         }
+    }
 
-        // process sub-areas in this support infill area with different densities
-        if (default_support_line_distance <= 0
-            || part.infill_area_per_combine_per_density.empty())
-        {
-            continue;
-        }
+    return added_something;
+}
 
-        for (unsigned int combine_idx = 0; combine_idx < part.infill_area_per_combine_per_density[0].size(); ++combine_idx)
-        {
-            const coord_t support_line_width = default_support_line_width * (combine_idx + 1);
+bool FffGcodeWriter::processSupportInfill(const SliceDataStorage& storage, LayerPlan& gcode_layer, const SupportInfillPart& part, const SupportConfig& config) const
+{
+    bool added_something = false;
 
-            Polygons support_polygons;
-            Polygons support_lines;
-            for (unsigned int density_idx = part.infill_area_per_combine_per_density.size() - 1; (int)density_idx >= 0; density_idx--)
-            {
-                if (combine_idx >= part.infill_area_per_combine_per_density[density_idx].size())
-                {
-                    continue;
-                }
+    // process sub-areas in this support infill area with different densities
+    if (config.line_distance <= 0 || part.infill_area_per_combine_per_density.empty())
+        return added_something;
```

Comment:
Code style: brackets around indented code block.
```suggestion
    {
        return added_something;
    }
```